presto-main/src/main/java/com/facebook/presto/operator/PartitionedOutputOperator.java

```diff
@@ -229,10 +232,8 @@ public PartitionedOutputOperator(
                 maxMemory);
 
         operatorContext.setInfoSupplier(this::getInfo);
-        // TODO: We should try to make this more accurate
-        // Recalculating the retained size of all the PageBuilders is somewhat expensive,
-        // so we only do it once here rather than in addInput(), and assume that the size will be constant.
-        operatorContext.newLocalSystemMemoryContext().setBytes(this.partitionFunction.getRetainedSizeInBytes());
+        this.systemMemoryContext = operatorContext.newLocalSystemMemoryContext();
```

Comment:
Do we wanna attribute this part of memory to user memory given it is still within an operator?

Reply:
For now Iet's don't change the pools this operator is allocating from. It will be taken care of when we unify the pools.